### ISSUE TITLE
CollectionView shows ALL indexed documents instead of only the selected

### ISSUE THREAD

**Issue by csidirop** [kitodo/kitodo-presentation#1106]
## Description

Setting the collection plugin without a ListView causes all documents to be shown instead of just the selected ones:

Left: Collection -> Listview -> PageView
Right: Collection -> PageView
![grafik](https://github.com/kitodo/kitodo-presentation/assets/43964592/937e6816-5aed-4e2a-914e-1548bf47793b)

This error exists even before the TYPO3 v11 merge.

## Reproduction

Steps to reproduce the behaviour:

1. Index some documents
2. Create two pages with Collection Plugins, one with a ListView and one with a PageView plugin
3. Setup the first Collection plugin to use the site with the PageView directly, the second one uses the ListView which then uses the PageView
4. See behavior

## Expected Behavior

Only documents in given collection should be shown, exactly like in the ListView.

## Screenshots and Examples

see above

## Environment

 -   OS version: Debian (in Docker)
 -   RDBMS version: Maria DB (MySQL 5.5.5-10.11.2-MariaDB-1:10.11.2+maria~ubu2204)
 -   Apache Solr version: -
 -   TYPO3 version: 10.4.38-dev
 -   PHP version: 7.4.33

**Comment by csidirop** [comment]
I searched a bit and found out the following: 

If we compare the `$collection` var from

https://github.com/kitodo/kitodo-presentation/blob/0e0c77f1d9fcc8a236eacb3e191ad46e133bb45c/Classes/Controller/ListViewController.php#L78

and 

https://github.com/kitodo/kitodo-presentation/blob/0e0c77f1d9fcc8a236eacb3e191ad46e133bb45c/Classes/Controller/CollectionController.php#L168

we see, that they are identical except that the first one has its `Kitodo\Dlf\Domain\Model\Collection` object in an array. If we encapsulating the collection object in an array `$collectionArray[] = $collection;` and replacing all references only the correct documents are shown:

![grafik](https://github.com/kitodo/kitodo-presentation/assets/43964592/1c03c3bf-792d-4935-9f38-97e3d37ec057)

Obviously this is just a workaround but I couldn't find the source of the call and why it differs. Maybe you can help me out.

**Comment by csidirop** [comment]
In the showAction function the search parameter are build from the url query parameter:
https://github.com/kitodo/kitodo-presentation/blob/0e0c77f1d9fcc8a236eacb3e191ad46e133bb45c/Classes/Controller/CollectionController.php#L170

But in this case `searchParameter` and `page` do not exists:

**Collection -> ListView -> PageView:**
`.../listview?tx_dlf_listview[action]=main&tx_dlf_listview[controller]=ListView&tx_dlf_listview[page]=1&tx_dlf_listview[searchParameter][collection]=1`

```array(4) {
  ["action"]=>
  string(4) "main"
  ["controller"]=>
  string(8) "ListView"
  ["page"]=>
  string(1) "1"
  ["searchParameter"]=>
  array(1) {
    ["collection"]=>
    string(1) "1"
  }
}
```

**Collection -> PageView:**
`.../sammlungen?tx_dlf_collection[action]=show&tx_dlf_collection[collection]=6&tx_dlf_collection[controller]=Collection`

```
array(3) {
  ["action"]=>
  string(4) "show"
  ["collection"]=>
  string(1) "6"
  ["controller"]=>
  string(10) "Collection"
}
```

**Comment by beatrycze-volk** [comment]
> Left: Collection -> Listview -> PageView
> Right: Collection -> PageView

Could you show me how look your configurations for both cases?

**Comment by csidirop** [comment]
- "Direct" way: (right example)
  - Page with `Kitodo: Pageview` plugin (in my case the default from DFG-Viewer)
  - Page with `Kitodo: Collection` plugin
    - Plugin config: Everything default, Set `Target page (with "Kitodo: Page View" plugin)` to point to the page with the viewer plugin
    ![grafik](https://github.com/kitodo/kitodo-presentation/assets/43964592/e46ec50c-441d-40f0-a836-98202fcd8ab6)
- "Indirect" way: (left example)
  - Page with `Kitodo: Pageview` plugin (in my case the default from DFG-Viewer)
  - Page with `Kitodo: ListView` plugin
    - Plugin config: Everything default, Set `Target page (with "Kitodo: Page View" plugin)` to point to the viewer page
  - Page with `Kitodo: Collection` plugin
    - Plugin config: Everything default, Set ` Target page (with "Kitodo: List View" plugin)` ` to point to the page containing the ListView plugin
    ![grafik](https://github.com/kitodo/kitodo-presentation/assets/43964592/0fb7d39f-eacb-4abd-8473-17d88e7cd4b5)

Here is the page tree export (changed extension to txt for github), it may help: [PageTree.xml.txt](https://github.com/kitodo/kitodo-presentation/files/13872747/PageTree.xml.txt).

Tell me if you need more!

**Comment by beatrycze-volk** [comment]
Thank you very much! It was really helpful :)

I was actually able to pinpoint the reason why it happens and why placing the `Collection` object into array solves it. The reason lies here:
https://github.com/kitodo/kitodo-presentation/blob/770df5aee7889f248d4de6693dbe1e447253a310/Classes/Common/Solr/SolrSearch.php#L439
The `filterquery` is empty when only single collection not stored in array is passed as parameter. Code inside the `foreach` loop is not executed and then `$collectionsQueryString` stays empty. As it is empty the `filterquery` can't be created from empty string: https://github.com/kitodo/kitodo-presentation/blob/770df5aee7889f248d4de6693dbe1e447253a310/Classes/Common/Solr/SolrSearch.php#L463

So indeed creating an `array` out of this single `Collection` object solves the problem.

**Comment by csidirop** [comment]
That sounds great! Im glad I could help you finally sorting this out :)

**Comment by sebastian-meyer** [comment]
Fixed by #1108
